File: eCommerce/UI/ecommerce_mobile/lib/model/transakcija.dart
import 'package:ecommerce_mobile/model/kategorija_transakcije.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'transakcija.g.dart';

@JsonSerializable()
class Transakcija {
  int? transakcijaId;
  int? korisnikId;
  User? korisnik;
  int? iznos;
  DateTime? datumTransakcije;
  String? opis;
  int? kategorijaTransakcijeId;
  KategorijaTransakcije? kategorijaTransakcije;
  String? status;


  Transakcija({
    this.transakcijaId,
    this.korisnikId,
    this.korisnik,
    this.iznos,
    this.datumTransakcije,
    this.opis,
    this.kategorijaTransakcijeId,
    this.kategorijaTransakcije,
    this.status,
  });

  factory Transakcija.fromJson(Map<String, dynamic> json) => _$TransakcijaFromJson(json);

  Map<String, dynamic> toJson() => _$TransakcijaToJson(this);
} 

File: eCommerce/eCommerce.Services/Database/Transakcija.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class Transakcija
    {
        public int TransakcijaId { get; set; }
        public int KorisnikId { get; set; }
        public User Korisnik { get; set; }
        public int Iznos { get; set; }
        public DateTime DatumTransakcije { get; set; }
        public string Opis { get; set; }
        public int KategorijaTransakcijeId { get; set; }

        public KategorijaTransakcije KategorijaTransakcije { get; set; }

        public StatusTransakcije Status { get; set; }
        public enum StatusTransakcije
        {
            Planirano=1,
            Gotovo=2,
            Otkazano=3
        }
    }
}
